JobRunner/DockerRunner.py

```diff
@@ -21,30 +18,41 @@ def __init__(self, logger=None):
         self.logger = logger
         self.containers = []
         self.threads = []
+        self.log_interval = 1
 
     def _sort_logs(self, sout, serr):
         """
         This is an internal function to sort and interlace output for NJS.
         This is not fully implemented yet and sould be rethought.
         """
-        # TODO: Fix sorting
 
-        lines = []
+        tslines = dict()
         if len(sout) > 0:
+            ierr = 0
             for line in sout.decode("utf-8").split('\n'):
                 if len(line) > 0:
-                    lines.append({'line': line, 'is_error': 0})
+                    (ts, txt) = line.split(maxsplit=1)
+                    if ts not in tslines:
+                        tslines[ts] = []
+                    tslines[ts].append({'line': txt, 'is_error': ierr})
         if len(serr) > 0:
+            ierr = 1
             for line in serr.decode("utf-8").split('\n'):
                 if len(line) > 0:
-                    lines.append({'line': line, 'is_error': 1})
-        return lines
+                    (ts, txt) = line.split(maxsplit=1)
+                    if ts not in tslines:
+                        tslines[ts] = []
+                    tslines[ts].append({'line': txt, 'is_error': ierr})
+        nlines = []
+        for ts in sorted(tslines.keys()):
+            nlines.extend(tslines[ts])
+        return nlines
 
     def _shepherd(self, c, job_id, subjob, queues):
```

Comment:
Could we come up with a better name for this function? Maybe something like log_shepherd or container_log_manager ?

Reply:
It looks like subjob is not used.